sql/expression/function/regexp_like.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+)
+
+// RegexpLike implements the REGEXP_LIKE function.
+// https://dev.mysql.com/doc/refman/8.0/en/regexp.html#function_regexp-like
+type RegexpLike struct {
+	Text    sql.Expression
+	Pattern sql.Expression
+	Flags   sql.Expression
+
+	cacheable bool
+	re        *regexp.Regexp
+}
+
+var _ sql.FunctionExpression = (*RegexpLike)(nil)
+
+// NewRegexpLike creates a new RegexpLike expression.
+func NewRegexpLike(args ...sql.Expression) (sql.Expression, error) {
+	var r *RegexpLike
+	switch len(args) {
+	case 3:
+		r = &RegexpLike{
+			Text:    args[0],
+			Pattern: args[1],
+			Flags:   args[2],
+		}
+	case 2:
+		r = &RegexpLike{
+			Text:    args[0],
+			Pattern: args[1],
+		}
+	default:
+		return nil, sql.ErrInvalidArgumentNumber.New("regexp_like", "2 or 3", len(args))
+	}
+	if canBeCached(r.Pattern) {
```

Comment:
Can't do this now, you have to wait until execution so the expressions are all resolved

Reply:
I retrofitted this from the `regexp_matches` code, I should have vetted this a bit more. Glad you caught it.